### DISCUSSION TITLE
CSS選擇器，如何選中深層標籤

### DISCUSSION THREAD

**Discussion by TehoChang** [Lidemy/forum#199] [前端相關問題]
CSS的『空格』選擇器只能往下選到一層，卻不能多層使用，例如
![截圖 2021-08-31 下午9 11 37](https://user-images.githubusercontent.com/61130008/131508445-a9734c0b-8ed0-4bbc-a976-cd447b77b41f.png)
測試結果如下：
![截圖 2021-08-31 下午9 18 19](https://user-images.githubusercontent.com/61130008/131509396-d77d392b-5624-45d5-8de0-7392633e8799.png)

**請問怎麼樣的寫法，可以把這三層都描述進去，並成功選到目標元素呢？？**

你可能會問說，寫比較少的選擇器就能選到，為什麼要多寫？
使用UI library的組件時，常常無法在程式碼中，直接在目標元素加上class / id，比方說一個table的component，無法直接在<td><th>上加上class/id。為了精準選到自己要修改的標籤，能多層深入寫選擇器似乎可以解決此問題？
還是其實有更高明修改UI庫組件外觀的方法 XD。

以上問題，謝謝～

**Comment by aszx87410** [comment]
之後問這種 CSS 類型的問題可以用 codepen 做出一個範例，其他人會比較好 debug
然後確認問題前可以先用 devtool 看一下 HTML 結構
你提供的範例之所以不能跑不是因為 CSS，是因為 div 不能放在 p 裡面，所以結構跟你想的不一樣
空格的那個 selector 應該是可以多層的

![螢幕快照 2021-08-31 下午9 35 22](https://user-images.githubusercontent.com/2755720/131512135-f6efcc1b-3aab-4888-b861-86acffe66cb4.png)
